src/app/shared/components/select/select.component.ts

```diff
@@ -0,0 +1,76 @@
+import { Output, EventEmitter, Component, Input, forwardRef } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR, ValidationErrors } from '@angular/forms';
+
+interface IOptionItem {
+  value: string,
+  text: string
+}
+
+@Component({
+  // tslint:disable-next-line:component-selector
+  selector: 'form-select',
+  templateUrl: './select.component.html',
+  styleUrls: ['./select.component.scss'],
+  providers: [{provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => SelectComponent), multi: true}]
+})
+export class SelectComponent implements ControlValueAccessor{
+
+  @Input() icon: string = ''
+  @Input() label: string = ''
+  @Input() options: IOptionItem[] = []
+  @Input() errorVisible = false
+  @Input() errors: ValidationErrors
+  @Input() errorsDef: any
+  @Output() valueChanged = new EventEmitter<string>()
+
+  private innerValue: any
+  private disabled: boolean = false
```

Comment:
should set it public, because you support set get without any logic

Reply:
ok @thienkieu